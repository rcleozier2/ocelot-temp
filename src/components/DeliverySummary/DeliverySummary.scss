@import "../../App-Variables.scss";

.delivery-summary {
    background-color: #ffffff;
    padding: 30px;
    border: 1px solid #eee;
    text-align: center;
    justify-content: center;
    align-items: center;
    filter: drop-shadow(2px 2px 4px #dedede);
    min-height: 250px;

    &-stats {
        font-size: 32px;
        color: $medly-blue;
        text-decoration: underline;
    }

    &-percentage {
        font-size: 20px;
        color: $medly-green;
    }
}