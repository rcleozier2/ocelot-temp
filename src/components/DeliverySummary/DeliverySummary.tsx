import React from "react";

import "./DeliverySummary.scss";

const DeliverySummary = (props: any) => {
    const completionPercentage = Math.round((props.tasks.total.completed / props.tasks.total.scheduled) * 100);
  return (
    <>
      <div className="delivery-summary">
        <div className="row">
          <div className="col-12">
            <p className="delivery-summary-stats">
              {props.tasks.total.completed} / {props.tasks.total.scheduled}
            </p>
            <p className="delivery-summary-percentage">
                {completionPercentage}% Complete
            </p>
          </div>
          <div className="col-12"> Sucessful Deliveries Today </div>
        </div>
      </div>
    </>
  );
};

export default DeliverySummary;
