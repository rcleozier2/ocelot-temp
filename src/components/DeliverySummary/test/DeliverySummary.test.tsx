import React from "react";
import DeliverySummary from "../DeliverySummary";
import renderer from "react-test-renderer";

it("renders DeliverySummary correctly", () => {
  const props = {
    tasks: {
      total: {
        completedOntime: 2,
        completedLate: 3,
        completed: 5,
        failedOntime: 2,
        failedLate: 2,
        failed: 4,
        scheduled: 9,
        justSends: 4
      }
    }
  };

  const component = renderer.create(<DeliverySummary {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});
