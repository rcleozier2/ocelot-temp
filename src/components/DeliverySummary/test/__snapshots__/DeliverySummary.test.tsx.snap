// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DeliverySummary correctly 1`] = `
<div
  className="delivery-summary"
>
  <div
    className="row"
  >
    <div
      className="col-12"
    >
      <p
        className="delivery-summary-stats"
      >
        5
         / 
        9
      </p>
      <p
        className="delivery-summary-percentage"
      >
        56
        % Complete
      </p>
    </div>
    <div
      className="col-12"
    >
       Sucessful Deliveries Today 
    </div>
  </div>
</div>
`;
